src/Razor/src/Microsoft.AspNetCore.Razor.VSCode.Extension/syntaxes/aspnetcorerazor.tmLanguage.yml

```diff
@@ -484,6 +486,34 @@ repository:
       - include: '#razor-codeblock-body'
     end: (?<=})
 
+  #>>>>> @if (...) { ... } <<<<<
+
+  if-statement:
+    name: 'meta.statement.if.razor'
+    begin: '(@)(if)\s*(?=\()'
+    beginCaptures:
+      1: { patterns: [ include: '#transition' ] }
+      2: { name: 'keyword.control.conditional.if.cs' }
+    patterns:
+      - include: '#csharp-condition'
+      - include: '#csharp-code-block'
+      - include: '#razor-codeblock-body'
+    end: (?<=})
+
+  #>>>>> @else [if (...)] { ... } <<<<<
+
+  else-part:
+    name: 'meta.statement.else.razor'
+    begin: '(?:^|(?<=}))\s*(else)\s*?(?: (if))?\s*?(?=[\n\(\{])'
```

Comment:
So we do a look-behind to ensure that there's a `}` behind us, but we don't have any guarantees it's the `}` of an `if`. If I'm recalling from previous questions I've had, we rely on the actual C# parsing to give the red squiggles in that case, yes?

Reply:
Yup right on!